jobflow.ipynb

```diff
@@ -1 +1,799 @@
+     },
+     "metadata": {}
+    }
+   ],
+   "source": [
+    "from jobflow import job, Response, Flow, run_locally\n",
+    "\n",
+    "@job\n",
+    "def get_ev_curve(structure, strain_lst):\n",
+    "    structures=generate_structures(structure,strain_lst=strain_lst)\n",
+    "    jobs = []\n",
+    "    volumes = []\n",
+    "    energies = []\n",
+    "    for istructure in range(len(strain_lst)):\n",
+    "        new_job = StaticQEMaker().make(structures[istructure])\n",
+    "        jobs.append(new_job)\n",
+    "        volumes.append(new_job.output.volume)\n",
+    "        energies.append(new_job.output.energy)\n",
+    "    return Response(replace=Flow(jobs, output={\"energies\": energies, \"volumes\": volumes}))\n",
+    "    \n",
+    "@job\n",
+    "def plot_energy_volume_curve_job(volume_lst, energy_lst):\n",
+    "    plot_energy_volume_curve(volume_lst=volume_lst, energy_lst=energy_lst)\n",
+    "\n",
+    "structure = bulk('Al', a=4.15, cubic=True)\n",
+    "relax = BaseQEMaker().make(structure=MSONAtoms(structure))\n",
+    "ev_curve_data = get_ev_curve(relax.output.structure, strain_lst=np.linspace(0.9, 1.1, 5))\n",
+    "# structure optimization job and (E, V) curve data job connected via relax.output\n",
+    "plot_curve = plot_energy_volume_curve_job(volume_lst=ev_curve_data.output[\"volumes\"], energy_lst=ev_curve_data.output[\"energies\"])\n",
+    "# (E, V) curve data job and plotting the curve job connected via ev_curve_data.output\n",
+    "qe_flow = [relax, ev_curve_data, plot_curve]  \n",
+    "# qe_flow is the QE flow that consists of the job for structural optimization, calculating the (E, V) curve data points and plotting the curve\n",
+    "run_locally(Flow(qe_flow), create_folders=True)  # order of the jobs in the flow is determined by connectivity"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "start_time": "2024-04-04T12:24:48.120617Z"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "The Mermaid diagram looks like this:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "outputs": [],
+   "source": [
+    "graph = to_mermaid(qe_flow, show_flow_boxes=True)\n",
```

Comment:
Could someone who has QE set up, run these lines please?

Reply:
You can simply install it via conda. I have done this as well

Reply:
Oh cool, didn't know this 😁

Reply:
We have a wide range of opensource codes available on conda-forge. You can find the full list here - at least the packages I maintain https://github.com/jan-janssen/conda-forge-contribution . If you are interested to add any other code I am happy to help.